src/embedding.rs

```diff
@@ -17,6 +17,10 @@ use std::sync::Arc;
 /// There can be much more but log the first few.
 const LOGGED_NUMBER_OF_BROKEN_ENTITIES: usize = 20;
 
+// FIXME: CachePadded
+pub type Row = crossbeam::utils::CachePadded<Vec<f32>>;
+pub type Matrix = Vec<Row>;
```

Comment:
Actually inner vector in Vec<Vec<f32>> is column so maybe change Row -> Col and write some comment for clarification.